server/src/main/java/org/elasticsearch/action/admin/cluster/desirednodes/TransportUpdateDesiredNodesAction.java

```diff
@@ -97,7 +97,7 @@ protected void masterOperation(
     @Override
     protected void doExecute(Task task, UpdateDesiredNodesRequest request, ActionListener<UpdateDesiredNodesResponse> listener) {
         final var minNodeVersion = clusterService.state().nodes().getMinNodeVersion();
-        if (request.isCompatibleWithVersion(minNodeVersion) == false) {
+        if (request.isCompatibleWithVersion(minNodeVersion.transportVersion) == false) {
```

Comment:
I'm not sure if this is requesting the node version or transport version.

Reply:
@fcofdez what does the node version here actually mean? Is it the version of the code running on the machine, or the transport/serialization version?

Reply:
The request contains a Node version (this api is for cloud informing ES of upcoming topology changes). However in this case, UpdateDesiredNodesRequest.isCompatibleWithVersion is checking the version for transport compatibility of the request itself.

Reply:
It's the oldest version among all the running cluster nodes. I think the change is correct, it was meant to prevent certain kind of changes during upgrades.

Reply:
OK, thanks both